src/cpp/polars/Series.cpp

```diff
@@ -174,9 +174,30 @@ namespace polars {
 
 // Location.
 
-// by  position of the indices
+    Series Series::iloc(int from, int to, int step) const {
+
+        if(empty()){
+            return Series();
+        }
+
+        arma::uvec pos;
+
+        if((from < 0) && (to > 0)){
+            auto effective_from =  values().size() + from;
+            pos = arma::regspace<arma::uvec>(effective_from, step, to - 1);
+        }
+        else if(from < 0 && to < 0){
+            auto effective_from = values().size() + from;
+            auto effective_to = values().size() + to;
+            pos = arma::regspace<arma::uvec>(effective_from, step, effective_to - 1);
+        }
+        else {
+            pos = arma::regspace<arma::uvec>(from,  step,  to-1);
+        }
```

Comment:
Ditto here

Reply:
See previous.

Reply:
Seems a bit weird to do from and to together since the logic seems independent. Also seems to combine make `else` implicitly `from >=0 || to == 0`, but it's implementation looks more like it would match `from >=0 && to > 0`.

Have you tested that this works correctly for a positive `from` and a negative `to`? Have you tested negative `from` and `to == 0`

Reply:
Have separated the logic and added tests that explicity look at all combinations of values for from and to.